azurelinuxagent/ga/cgroupcontroller.py

```diff
@@ -142,21 +143,31 @@ def _get_parameters(self, parameter_name, first_line_only=False):
     def is_active(self):
         """
         Returns True if any processes belong to the cgroup. In v1, cgroup.procs returns a list of the thread group IDs
-         belong to the cgroup. In v2, cgroup.procs returns a list of the process IDs belonging to the cgroup.
+        belong to the cgroup. In v2, cgroup.procs returns a list of the process IDs belonging to the cgroup.
         """
         try:
-            procs = self._get_parameters("cgroup.procs")
-            if procs:
-                return len(procs) != 0
-        except (IOError, OSError) as e:
-            if e.errno == errno.ENOENT:
-                # only suppressing file not found exceptions.
-                pass
-            else:
-                logger.periodic_warn(logger.EVERY_HALF_HOUR,
-                                     'Could not get list of procs from "cgroup.procs" file in the cgroup: {0}.'
-                                     ' Internal error: {1}'.format(self.path, ustr(e)))
-        except CGroupsException as e:
+            def _found_cgroup_procs(file):
+                try:
+                    procs = fileutil.read_file(file).splitlines()
+                    if len(procs) > 0:
+                        return True
+                except (IOError, OSError) as e:
+                    if e.errno == errno.ENOENT:
+                        # only suppressing file not found exceptions.
+                        pass
+                    else:
+                        raise
+                return False
+
+            # In v1, the cgroup.procs file is present in the service/slice cgroup directory.
+            if _found_cgroup_procs(os.path.join(self.path, "cgroup.procs")):
+                return True
+
+            # In v2, the cgroup.procs file is present in the scope cgroup for extensions
+            for cgroup_file in glob.iglob(os.path.join(self.path, "*/cgroup.procs")):
```

Comment:
what is the * matching?

Reply:
* matches sub-directories, here it will be child cgroups

ex:

/sys/fs/cgroup/azure.slice/azure-vmextensions.slice/azure-vmextensions-microsoft.azure.customscript.slice/**enable_123.scope**/cgroup.procs

Reply:
I think we can scope it to *.scope/cgroup.procs